src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/AutoFlushingStreamTest.cs

```diff
@@ -0,0 +1,179 @@
+            var fileContents = new byte[INIT_BYTES];
+            var randomGenerator = new Random();
+            randomGenerator.NextBytes(fileContents);
+            await clientStream.WriteAsync(fileContents, 0, INIT_BYTES);
+            await clientStream.FlushAsync();
+            await serverStream.WriteAsync(fileContents, 0, INIT_BYTES);
+            await serverStream.FlushAsync();
+
+            // Act
+            var serverReads = Task.Factory.StartNew(() =>
+            {
+                for (var i = 0; i < 100000; i++)
+                {
+                    var tmpBuffer = new byte[10];
+                    try
+                    {
+                        var result = autoFlushingStream.Read(tmpBuffer, 0, 10);
+                        Assert.Equal(10, result);
+                    }
+                    catch (Exception e)
+                    {
+                        Assert.True(false);
+                    }
+                }
+            });
+            var serverWrites = Task.Factory.StartNew(() =>
+            {
+                for (var i = 0; i < 100000; i++)
+                {
+                    try
+                    {
+                        var tmpBuffer = new byte[10];
+                        randomGenerator.NextBytes(tmpBuffer);
+                        autoFlushingStream.Write(tmpBuffer, 0, 10);
+                    }
+                    catch (Exception e)
+                    {
+                        Assert.True(false);
+                    }
+                }
+            });
+
+            var clientReads = Task.Factory.StartNew(async () =>
+            {
+                for (var i = 0; i < 100000; i++)
+                {
+                    try
+                    {
+                        var tmpBuffer = new byte[10];
+                        var result = await clientStream.ReadAsync(tmpBuffer, 0, 10);
+                        Assert.Equal(10, result);
+                    }
+                    catch (Exception e)
+                    {
+                        Assert.True(false);
+                    }
+                }
+            });
+            var clientWrites = Task.Factory.StartNew(async () =>
+            {
+                for (var i = 0; i < 100000; i++)
+                {
+                    try
+                    {
+                        var tmpBuffer = new byte[10];
+                        randomGenerator.NextBytes(tmpBuffer);
+                        await clientStream.WriteAsync(tmpBuffer, 0, 10);
+                        await clientStream.FlushAsync();
+                    }
+                    catch (Exception e)
+                    {
+                        Assert.True(false);
+                    }
+                }
+            });
+
+            // Assert
+            Task.WaitAll(new[] { serverReads, serverWrites, clientReads, clientWrites });
+        }
+
+        [Fact]
+        public async void ParallelReadWrite_ClientServer_Record()
```

Comment:
Same test as above, but using `Record.Exception` assertions instead. Will only keep one of these tests, just want to ensure it can appropriately fail in the normal test run context.